lib/std/mem.zig

```diff
@@ -1534,6 +1534,29 @@ test "writeIntBig and writeIntLittle" {
     try testing.expect(eql(u8, buf2[0..], &[_]u8{ 0xfc, 0xff }));
 }
 
+/// Swap the byte order of all the members of the fields of a struct
+/// (Changing their endianess)
+pub fn bswapAllFields(comptime S: type, ptr: *S) void {
+    if (@typeInfo(S) != .Struct) @compileError("bswapAllFields expects a struct as the first argument");
+    inline for (std.meta.fields(S)) |f| {
+        @field(ptr, f.name) = @byteSwap(f.field_type, @field(ptr, f.name));
+    }
+}
+
+test "bswapAllFields" {
+    var s: Elf32_Chdr = .{
```

Comment:
`Elf32_Chdr` is not defined here, I'd replace this test with something like this:
```zig
const T = extern struct {
    f0: u8,
    f1: u16,
    f2: u32,
};
var s = T{
    .f0 = 0x12,
    .f1 = 0x1234,
    .f2 = 0x12345678,
};
bswapAllFields(T, &s);
try std.testing.expectEqual(T{
    .f0 = 0x12,
    .f1 = 0x3412,
    .f2 = 0x78563412,
}, s);
```